packages/datadog-instrumentations/src/tedious.js

```diff
@@ -0,0 +1,65 @@
+'use strict'
+
+const {
+  channel,
+  addHook,
+  AsyncResource
+} = require('./helpers/instrument')
+const shimmer = require('../../datadog-shimmer')
+
+addHook({ name: 'tedious', versions: [ '>=1.0.0' ] }, tedious => {
+  const startCh = channel('apm:tedious:makeRequest:start')
+  const asyncEndCh = channel('apm:tedious:makeRequest:async-end')
+  const endCh = channel('apm:tedious:makeRequest:end')
+  const errorCh = channel('apm:tedious:makeRequest:error')
+  shimmer.wrap(tedious.Connection.prototype, 'makeRequest', makeRequest => function (request) {
+    const asyncResource = new AsyncResource('bound-anonymous-fn')
```

Comment:
This creates an unneeded async resource if the below condition is true and the function returns early.